listenbrainz/db/user.py

```diff
@@ -572,3 +572,30 @@ def search(search_term: str, limit: int, searcher_id: int = None) -> List[Tuple[
         else:
             search_results = [(row.musicbrainz_id, row.query_similarity, None) for row in rows]
         return search_results
+
+
+def search_user_name(search_term: str, limit: int) -> List[object]:
+
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+           SELECT musicbrainz_id, similarity(musicbrainz_id, :search_term) AS query_similarity
+              FROM "user"
+             WHERE musicbrainz_id <% :search_term
+          ORDER BY query_similarity DESC
+             LIMIT :limit
+            """), {
+            "search_term": search_term,
+            "limit": limit
+        })
+
```

Comment:
Looking at this now again now that I understand what is happening better and where the code comes from, can't we just use the existing `search` defined above? This code seems to just duplicate it for the most part (ignoring, unless i'm missing something.

Reply:
Sounds good, to refactor the common part in a separate function.